[Wang Pei] iP

wangpalex opened the pull request:
## Duke v1.0
> "I don't know what to quote so I just quote myself." - myself 🤡 

### Duke v1.0 has already implemented basic funcionalities of a ToDo list:
- ~~nice~~ *ordinary* UI.
- track tasks, deadlines and events as you want.
- mark tasks done whenever you want.
- auto-save the file after changes.
- load task list from given file.
- find the wanted task with keywords.

### You can find Duke v1.0. customized version [here](https://github.com/wangpeialex/ip)

### Command of Duke follows the format below:
1. todo `taskName` [-p `priority`]
2. deadline `deadlineName` /by `dateTime: yyyy/MM/dd HH:mm` [-p `priority`]
3. event `eventName` /at `startDateTime: yyyy/MM/dd HH:mm` `endDateTime: yyyy/MM/dd HH:mm` [-p `priority`]
4. list
5. done `index`
6. delete `index`
7. find `keyword`

### Features under working:
- [x] A-Gradle
- [x] A-Checkstyle
- [x] Level-10
- [x] A-Varargs

### If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Added basic conversation function

Commit:
Implemented list for duke to store strings

Commit:
Implemented mark as done function

Commit:
Implemented Deadline and Event classes, refactored structure

Commit:
Implemented some error handling

Commit:
Implemented deletion of task

Commit:
Added tests

Commit:
Changed error output

Commit:
Include Junit 5 in the project

Commit:
Implement save function of todo list

Commit:
Add file for saving to .gitignore

Commit:
Implement DateTime in Deadline and Event class and their parsing

Commit:
Merge branch 'master' into branch-Level-7

Commit:
Merge branch 'branch-Level-7'

Commit:
Add explanation for branch-Level-8

Commit:
Add package structure to the project

Commit:
Restructure project to be more OOP

Commit:
Revert "Restructure project to be more OOP"

This reverts commit d04560a9d7e3cfa5136da36578e70906d5131cd4.

Commit:
Revert "Revert "Restructure project to be more OOP""

This reverts commit 733058b49ffcc6b6b5a747db9f82393af2b19788.

Commit:
Implement save to file and load from file

Commit:
Add tests for Parser methods

Commit:
Add tests for TaskList methods

Commit:
Add more tests to TaskListTest

Commit:
Create Jar for the project

Commit:
Add JavaDoc for public classes and methods

Commit:
Add JavaDoc to Command class

Commit:
Add JavaDoc and modify coding standard

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Implement find function

Commit:
Add message in README.md

Commit:
Add Gradle to Duke project

Commit:
Add plugins to build.gradle

Commit:
Update Junit version

Commit:
Add checkstyle config files

Commit:
Add javaFX dependencies to Gradle

chrus-chong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,85 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
Should this boolean variable be named in a more intuitive manner? Perhaps `hasExited` might be better?

wangpalex replied:
Thanks for the suggestion! `hasExited` is indeed a better var name.

chrus-chong commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+     * Returns a boolean indicating if the main loop should be broke.
+     * @return A boolean indicating if the main loop should be broke
+     */
+    public boolean exeResult() {
```
Perhaps you may wish to name your `exeResult( )` method into something more intuitive? Was this meant to be shortform for `executeResult( )` ?

wangpalex replied:
Sure, I will use full description for this function name.

chrus-chong commented on src/main/java/duke/command/CommandFind.java:
```diff
@@ -0,0 +1,33 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        TaskList matched = tasks.find(keyword);
```
Should this variable be renamed to `matchedTasks` for better identification?

wangpalex replied:
Thanks for the suggestion!

chrus-chong commented on src/main/java/duke/command/CommandAdd.java:
```diff
@@ -0,0 +1,52 @@
+public class CommandAdd extends Command {
+    Task taskToAdd;
+
+    /**
+     * Instantiates an object with given task to be added when
+     * called to execute.
+     * @param taskToAdd
+     */
```
Consider adding a blank line between the method description and the `@param` statement to follow the Javadocs formatting in the [Java Coding Standard](https://nus-cs2103-ay1920s2.github.io/website/coding-standards/java/intermediate.html#variables).

wangpalex replied:
Hi, thanks for pointing out these defects so I can improve my code quality. I will make modifications according to suggestions! 😄

wangpalex replied:
Noted, I will check out the coding standard again.

chrus-chong reviewed (commented):
Great job on the project! I cannot seem to find any recurring major coding standard violations. Keep up the good work!

Commit:
Add GUI to Duke

Commit:
Fix mac JavaFX font problem

bernardwan commented on test/duke/util/ParserTest.java:
```diff
@@ -0,0 +1,65 @@
+
+    @Test
+    void parseDescription() throws DukeException {
+        assertEquals("description part 1 description part 2", Parser.parseDescription(new String[] {"taskType","description part 1", "description part 2"}));
```
Should have a line break perhaps

bernardwan commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+    public String deleteTask(int idx) throws DukeException {
+        try {
+            StringBuilder msg = new StringBuilder();
+            Task curr = this.tasks.remove(idx-1);
```
Under advanced coding standard violations, it is said not to use 'this' for fields if it is not necessary. Can consider removing this here and other parts of code perhaps.

wangpalex replied:
Hey @bernardwan, thanks for pointing that out, I will remove redandunt `this`.

bernardwan commented on src/main/java/duke/command/CommandDone.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    /**
+     * Execute the mark as done action and auto-saves to the file when called.
```
For header comments, should leave a line in between the Description of the method and the description of the params.

bernardwan reviewed (commented):
Naming seems very clear overall! Couldn't find much to comment on.

tsy24 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    /**
+     * Main logic of Duke,
```
Consider starting the sentence with a verb like `Starts` to follow the [Java Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html)

tsy24 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,99 @@
+     * @return      Execution message of the input or error messages.
+     * @throws DukeException when something goes wrong.
+     */
+
```
For Javadoc comments, the coding standard is to have an empty line between description and parameter section. Also, there should not be a blank line between the documentation block and the method.

wangpalex replied:
Thanks @tsy24, I would fix on that.

tsy24 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,73 @@
+
+    /**
+     * Constructor.
+     * Instantiates a Deadline object with given description, isDone status and
```
Could this be a typo? Did you mean Event object instead?

tsy24 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,84 @@
+     * Greets user.
+     * @param name User name to display.
+     */
+    public void greeting(String name) {
```
Perhaps you can consider naming your method with a verb? Maybe something like `greetUser` or `sayGreeting`.

tsy24 commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,269 @@
+     * @throws DukeException When the input is empty String or the command is invalid
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        String[] splited = input.strip().split(" +");
```
Plural form should be used on names representing a collection of objects. Perhaps `splitedStrings` would be a better name?

tsy24 commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,269 @@
+     * @throws DukeException when the index is missing from the input.
+     */
+    public static int parseIndex(String[] strArr) throws DukeException {
+        if(strArr.length < 2) {
```
According to the coding standard, Java reserved words should be followed by a white space, eg. `if (condition) {` and `for (i = 0; i < 10; i++) {` I noticed the same issue in several other places too.

tsy24 commented on test/duke/util/TaskListTest.java:
```diff
@@ -0,0 +1,49 @@
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Consider listing imported classes explicitly to follow the coding standards.

Commit:
Fix coding standard violations and improve code quality

Commit:
Change font size of send button

Commit:
Modify build.gradle

shadowJar is not producing right packaged application

We need to fix this in order to release Duke v1.0

Let's,
- update the mainClassName to Launcher

Commit:
Add assertion in addTask.

Commit:
Parser class: Add exception handling in decode method.

Commit:
Parser class: Refactor by extracting methods in decode() to improve code quality

The decode() method is very lengthy and hard to read.

We need to refactor it to improve readability.

Let's:
- Extract methods for decode().
- Define formatting string literals as constants.

Commit:
Refactor classes to define string literals as private constants

wangpalex committed:
Merge pull request #1 from wangpeialex/branch-A-Assertions

Branch a assertions

wangpalex committed:
Merge pull request #2 from wangpeialex/branch-A-CodeQuality

Branch a code quality

Commit:
Create Priority enum class

Commit:
Introduce priority to tasks

Commit:
Add comparing logic for tasks

Commit:
Add parse methods for priority

Commit:
Modify constructor of Tasks to include priority

Commit:
Add parser methods for priority

Commit:
Modify compare methods of tasks

Commit:
Add gradle.yml for CI

wangpalex committed:
Merge pull request #3 from wangpeialex/C-Priority

C priority

Commit:
Merge branch 'A-CI'

Commit:
Modify UI

Commit:
Fix event parsing error

wangpalex committed:
Set theme jekyll-theme-cayman

Commit:
Draft UG

Commit:
Merge branch 'master' of https://github.com/wangpeialex/ip

Commit:
Modify UG

Commit:
Fix coding standard violations

Commit:
Change version code
